profiling/server.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profiling
+
+import (
+	"net/http"
+	"net/http/pprof"
+	"strconv"
+	"sync"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	// profilingPort is the port where we expose profiling information if profiling is enabled
+	profilingPort = ":8008"
+
+	// profilingKey is the name of the key in config-observability config map that indicates whether profiling
+	// is enabled of disabled
+	profilingKey = "profiling.enable"
+)
+
+// Handler holds the main HTTP handler and a flag indicating
+// whether the handler is active
+type Handler struct {
+	enabled bool
+	handler http.Handler
+	log     *zap.SugaredLogger
+	mutex   sync.Mutex
```

Comment:
We usually group the mutex together with the value it guards, like:

```golang
enabledMux sync.Mutex
enabled bool
```

To make clear what it does or what it's supposed to do rather.